src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -104,6 +104,8 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
             await CreateServicesAsync(cancellationToken).ConfigureAwait(false);
 
             await CreateContainersAndExecutablesAsync(cancellationToken).ConfigureAwait(false);
+
+            distributedApplicationLogger.LogInformation("Distributed application started. Press CTRL-C to stop.");
```

Comment:
We shouldn't do this here. It should be inside of StartAsync in DistributedApplication.

Reply:
I think we are going to need to create a distributed application lifecycle service whose main goal is to dump out these informational messages at just the right time.

If we put it in StartAsync it means you'll see the app started message before the dashboard is actually running which might be confusing.

Reply:
I realize. Maybe this is a good enough hack for now as much as I don't like it.

Reply:
I can put the lifecycle service in this morning. It's easy to do and will potentially be useful for other things.

Reply:
You'd also need to add events right?

Reply:
Not sure what you mean. I was just going to inject a class that has IHostedLifecycleService.

I'd leave the listening on message where it is and move the two messages in this PR to it (we want the listening on message to appear even doing startup to get that browser launched ASAP).